pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -482,6 +482,12 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 		if err != nil {
 			return nil, err
 		}
+		defer func() {
+			if err := gce.deleteStaticIP(name, region); err != nil {
```

Comment:
don't we return the error to retry (in theory at least, i think it does retry though) today?